core/src/main/java/org/jruby/RubyHash.java

```diff
@@ -1581,28 +1575,59 @@ public IRubyObject each_key(final ThreadContext context, final Block block) {
         return block.isGiven() ? each_keyCommon(context, block) : enumeratorizeWithSize(context, this, "each_key", RubyHash::size);
     }
 
-    @JRubyMethod(name = "transform_keys")
-    public IRubyObject transform_keys(final ThreadContext context, final Block block) {
-        if (block.isGiven()) {
-            RubyHash result = newHash(context.runtime);
-            visitAll(context, new TransformKeysVisitor(block), result);
-            return result;
+    @JRubyMethod(name = "transform_keys", optional =  1, rest = true)
+    public IRubyObject transform_keys(final ThreadContext context, IRubyObject[] arg0, final Block block) {
+        RubyArray transformKeys = null;
+        RubyHash result = newHash(context.runtime);
+
+        if((arg0.length > 0) && (arg0 [0] != null))
+            transformKeys = ((RubyHash) arg0[0]).keys();
+
+        if(block.isGiven())
```

Comment:
All 'if(' should be 'if ('.  Also a couple of other style rules is if you have a single line if you can omit braces but only if it is on the same line.  Anything which also has else or you put on the next line should use braces.